### PR TITLE
Make network indicator helder thread-safe.

### LINKED ISSUE #203: Improve NetworkIndicator
- [ ] increments/decrements must be thread-safe/atomic
- [ ] `counter` should never be negative (currently possible)
- [ ] better function names - We do not always 'start' or 'end' the indicator.

### PR DESCRIPTION
**Pull request by mathebox**
- Allow only positive values for counter.

Fixes #203

### CODE REVIEW

File: xikolo-ios/Helpers/NetworkIndicator.swift [hunk 1]
```diff
@@ -11,19 +11,24 @@ import Spine
 
 class NetworkIndicator {
 
-    static var counter = 0
+    private static let queue = DispatchQueue(label: "de.xikolo.queue.network-indicator")
```
Comment (jprberlin) [inline xikolo-ios/Helpers/NetworkIndicator.swift]: should this be brand specific?

File: xikolo-ios/Helpers/NetworkIndicator.swift [hunk 2]
```diff
@@ -11,19 +11,24 @@ import Spine
...
+    private (set) static var counter = 0
 
     class func start() {
-        counter += 1
-        update()
+        self.queue.sync {
+            self.counter += 1
```
Comment (jprberlin) [inline xikolo-ios/Helpers/NetworkIndicator.swift]: is there ++ in Swift

File: xikolo-ios/Helpers/NetworkIndicator.swift (hunk 1 above)
Comment (mathebox) [reply xikolo-ios/Helpers/NetworkIndicator.swift]: There is no need for that.

File: xikolo-ios/Helpers/NetworkIndicator.swift (hunk 2 above)
Comment (mathebox) [reply xikolo-ios/Helpers/NetworkIndicator.swift]: No. This was removed in Swift 3.

**Pull request merged**